src/app.js

```diff
@@ -1,7 +1,20 @@
 /* @flow */
 'use strict'
 
+// SIGINT and uncaughtException do not run cleanup
+// so we have an exitHandler to clean up the database connection
+function exitHandler (error) {
+  if (error) {
+    console.error('Exiting: ', error)
+    process.exit(1)
+  }
+  process.exit(0)
+}
+
 module.exports = require('./lib/app')
 if (!module.parent) {
+  process.on('SIGINT', exitHandler)
+  process.on('uncaughtException', exitHandler)
```

Comment:
Definitely agree that this seems dangerous and hacky.

``` js
throw null
```

^ makes this PR exit with no error, no explanation and status code 0 (success).

Normal behavior is to print:

```
/opt/workspace/five-bells-ledger/test.js:1
(function (exports, require, module, __filename, __dirname) { throw null
                                                              ^
null
```

I'm not so much arguing that you need to handle `throw null` correctly, more illustrating why overriding the default exception handler is problematic. I'm trying to understand more about the problem that you're trying to solve, maybe there is a less hacky approach.